### PR TITLE
Scripts for parameterisation

### PR DESCRIPTION
**Pull request by alisonrclarke**
Added scripts to help with the parameterisation process:

 * Updated reframe parameterisation script to zip up results at the end of a run
 * Added script to fetch output files from Hamilton
 * Added script to plot correlations
 * Added script to combine the best results from all runs
 * Updated docs

### CODE REVIEW

File: hamilton/parameter_analysis/merge_best_results.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+if __name__ == "__main__":
+    dataframes = []
+
+    for d in os.listdir(sys.argv[1]):
+        full_path = os.path.join(sys.argv[1], d)
+        if os.path.isdir(full_path) and not d.endswith("corrupted"):
+            f = os.path.join(full_path, "lowest_to_highest_mses.csv")
+            if os.path.isfile(f):
+                df = pd.read_csv(f, sep=",")
+                df = df.loc[(df["mean_squared_error"] < 3.5)]
```
Comment (MarkLTurner) [inline hamilton/parameter_analysis/merge_best_results.py:15]: Could 3.5 be configurable through a variable, say, called quality_threshold that has a default value of 3.5?

File: hamilton/parameter_analysis/plot_correlations.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+if __name__ == "__main__":
+    dir = sys.argv[1]
+    filename = sys.argv[2]
+    file_path = os.path.join(dir, filename)
+    df = pd.read_csv(file_path)
+    y_vars = ["mean_squared_error"]
+
+    if filename.endswith("best_mses.csv"):
+        x_vars = list(df.columns.values)[0:-3]
```
Comment (MarkLTurner) [inline hamilton/parameter_analysis/plot_correlations.py:16]: I can't remember what these numbers refer to. Would it make sense to assign them to global variables?

**Review by MarkLTurner — DISMISSED** [review]
Great, I added only two small comments that could be ignored.

**Review by MarkLTurner — DISMISSED** [review]

File: hamilton/parameter_analysis/plot_correlations.py (hunk 2 above)
Comment (alisonrclarke) [reply hamilton/parameter_analysis/plot_correlations.py:19]: I've just added comments to explain the magic numbers; I think it's probably easier to understand in context than moving to globals out of context.

**Review by MarkLTurner — APPROVED** [review]

**Pull request merged**